### PR TITLE
Change the nn-lib to the latest version 2.9.0.

### PR DESCRIPTION
**Pull request by tanmaydas82**
BUG=[b.corp.google.com/issues/239852051](https://b.corp.google.com/issues/239852051)

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/add.cc [hunk 1]
```diff
@@ -40,9 +40,10 @@ TfLiteStatus Prepare(TfLiteContext* context, TfLiteNode* node) {
...
 }
 
-void EvalAdd(TfLiteContext* context, TfLiteNode* node, TfLiteAddParams* params,
-             const OpDataAdd* data, const TfLiteEvalTensor* input1,
-             const TfLiteEvalTensor* input2, TfLiteEvalTensor* output) {
+TfLiteStatus EvalAdd(TfLiteContext* context, TfLiteNode* node,
+                     TfLiteAddParams* params, const OpDataAdd* data,
+                     const TfLiteEvalTensor* input1,
+                     const TfLiteEvalTensor* input2, TfLiteEvalTensor* output) {
   // printf("MCP DEBUG EvalAdd\n");
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/xtensa/add.cc:47]: let's remove commented out code.

File: tensorflow/lite/micro/kernels/xtensa/sub.cc [hunk 2]
```diff
@@ -104,11 +148,54 @@ TfLiteStatus EvalSubQuantized(TfLiteContext* context, TfLiteNode* node,
...
+      const RuntimeShape extended_output_shape =
+          RuntimeShape::ExtendedShape(5, tflite::micro::GetTensorShape(output));
+      const int* input1_dims = extended_input1_shape.DimsData();
+      const int* input2_dims = extended_input2_shape.DimsData();
+      const int* output_dims = extended_output_shape.DimsData();
+      int b;
+      int inp1_off = 0;
+      int inp2_off = 0;
+      int out_off;
+      out_off =
```
Comment (advaitjain) [inline tensorflow/lite/micro/kernels/xtensa/sub.cc:170]: we should be naming these variables better. Let's add that to the cleanup/refactor bug and address in a follow-up PR.

**Review by advaitjain — CHANGES_REQUESTED** [review]
cleanup comments that we can punt for a follow-up PR. Let's make a bug and then we can merge this PR so that we can get the nnlib verison updated before coming back to refactor.

File: tensorflow/lite/micro/kernels/xtensa/add.cc (hunk 1 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/kernels/xtensa/add.cc:47]: Done

File: tensorflow/lite/micro/kernels/xtensa/add.cc [hunk 3]
```diff
@@ -56,13 +57,38 @@ void EvalAdd(TfLiteContext* context, TfLiteNode* node, TfLiteAddParams* params,
...
+    TF_LITE_ENSURE(context, err == 0);
+
+    err = xa_nn_vec_activation_min_max_f32_f32(
+        tflite::micro::GetTensorData<float>(output),
+        tflite::micro::GetTensorData<float>(output),
+        data->output_activation_min_f32, data->output_activation_max_f32,
+        flat_size);
+
+    TF_LITE_ENSURE(context, err == 0);
+#else
```
[the remark this answers is not in the document]
Comment (tanmaydas82) [reply tensorflow/lite/micro/kernels/xtensa/add.cc:82]: Done

File: tensorflow/lite/micro/kernels/xtensa/sub.cc [hunk 4]
```diff
@@ -81,12 +81,56 @@ TfLiteStatus EvalSubQuantized(TfLiteContext* context, TfLiteNode* node,
...
+          output_dims[1] * output_dims[2] * output_dims[3] * output_dims[4];
+      if (input1_dims[0] > 1) {
+        inp1_off =
+            input1_dims[1] * input1_dims[2] * input1_dims[3] * input1_dims[4];
+      }
+      if (input2_dims[0] > 1) {
+        inp2_off =
+            input2_dims[1] * input2_dims[2] * input2_dims[3] * input2_dims[4];
+      }
```
[the remark this answers is not in the document]
Comment (tanmaydas82) [reply tensorflow/lite/micro/kernels/xtensa/sub.cc:119]: Added a new bug https://b.corp.google.com/issues/259724572

File: tensorflow/lite/micro/kernels/xtensa/sub.cc [hunk 5]
```diff
@@ -81,12 +81,56 @@ TfLiteStatus EvalSubQuantized(TfLiteContext* context, TfLiteNode* node,
...
                       &op_params);
+#if !defined(HIFI4)
   bool need_broadcast = reference_ops::ProcessBroadcastShapes(
       tflite::micro::GetTensorShape(input1),
       tflite::micro::GetTensorShape(input2), &op_params);
+#endif  // !defined(HIFI4)
 
   switch (output->type) {
     case kTfLiteInt8: {
+#if defined(HIFI4)
```
[the remark this answers is not in the document]
Comment (tanmaydas82) [reply tensorflow/lite/micro/kernels/xtensa/sub.cc:94]: Added a new bug https://b.corp.google.com/issues/259724572

File: tensorflow/lite/micro/kernels/xtensa/sub.cc (hunk 2 above)
Comment (tanmaydas82) [reply tensorflow/lite/micro/kernels/xtensa/sub.cc:173]: Added a new bug https://b.corp.google.com/issues/259724572

**Review by advaitjain — APPROVED** [review]

**Pull request merged**